contents/code/tilelist.js

```diff
@@ -216,6 +216,25 @@ TileList.prototype.connectSignals = function(client) {
     });
     client.activeChanged.connect(function() {
         try {
+            // Make sure the newly active client is not covered by
+            // a maximized one.
+            if (client.setMaximize != null && client.active == true) {
```

Comment:
So the `setMaximize` is a function that KWin will provide in 5.16. Probably helpful to note that somewhere. I was looking for what happens if _a_ client doesn't have it, and then you wouldn't unmaximize the old maximized client, and...

But since it's API, this one either exists for all clients or none.

Reply:
Yes, it's merged for v5.16 and yes, if it's not implemented, well, then it does nothing. But it's the same now anyway.

Reply:
So, since this function does _nothing_ if setMaximize isn't a thing, and we've established that can't change, why not move the check before, and only do the `connect()` if setMaximize exists?

Reply:
hmm, clever. Will do.

Reply:
So, just to make sure. You suggest to connect the `activeChanged` signal twice - once for focus tracking and the second time for the maximization stuff if available?

Reply:
I actually missed the focus tracking stuff. But yeah, that sounds better - if caching is worth it, it should be done here as well. (Alternatively, check for setMaximize  before and close the variable)

Reply:
Done.